spring-vault-core/src/main/java/org/springframework/vault/authentication/KubernetesServiceAccountTokenFile.java

```diff
@@ -96,7 +97,7 @@ public KubernetesServiceAccountTokenFile(Resource resource) {
 			this.token = readToken(resource);
 		}
 		catch (IOException e) {
-			throw new VaultException(String.format(
+			throw new VaultClientException(String.format(
```

Comment:
`VaultException` should be fine here.

Reply:
It is, but ignores the reasons that VaultClientException was created, which we need to come to consensus on. It exists to clearly separate cause and effect
Cause: Something happened on the Client side while preparing a request = VaultClientException
Cause: Something happened while making the request = VaultRemoteException or VaultHttpException
Cause: Something happened while the request is done but parsing it initially: VaultRemoteException
Cause: It's definitely Vault Related but we've lost context - VaultException

That's my logic in a nutshell.

Reply:
We're catching `IOException` here for the sole purpose of wrapping it to a `RuntimeException`. There's nothing Vault specific here except that the code lives in Spring Vault.

Reply:
Seems like that argument, if taken to extremes says "Convert everything to a generic RuntimeException, dump VaultException and VaultClientException and just wrap the HttpStatusException in a generic RuntimeException" ;)

I am pretty sure you don't mean that.
I'm making the case that this is middle ware and generally most exceptions should be "refunneled" into some subclass of VaultException.